chain/default_syncer.go

```diff
@@ -302,42 +302,51 @@ func (syncer *DefaultSyncer) widen(ctx context.Context, ts types.TipSet) (types.
 	// Lookup tipsets with the same parents from the store.
 	parentSet, err := ts.Parents()
 	if err != nil {
-		return nil, err
+		return types.NoTipSet, err
```

Comment:
We have a bit of a convention to use 'Undef' to for nil values of known types (`address.Undef`, `cid.Undef`). `types.UndefTipSet` would help distinguish this from a flag or enum. Not a big deal though.